web/src/components/tasks/BusinessFormation.test.tsx

```diff
@@ -407,34 +407,124 @@ describe("<BusinessFormation />", () => {
       expect(subject.queryByTestId("dependency-alert")).not.toBeInTheDocument();
     });
 
-    it("navigates back to business tab from the contact tab", async () => {
+    it("navigates from business tab to payment tab and back to business tab", async () => {
       renderWithData({});
       await submitBusinessTab();
+      await submitContactsTab();
+      await submitReviewTab();
+
+      fireEvent.click(subject.getByText(BusinessFormationDefaults.previousButtonText));
+      await waitFor(() => {
+        expect(subject.getByTestId("review-section")).toBeInTheDocument();
+      });
 
       fireEvent.click(subject.getByText(BusinessFormationDefaults.previousButtonText));
+      await waitFor(() => {
+        expect(subject.queryByTestId("contacts-section")).toBeInTheDocument();
+      });
 
+      fireEvent.click(subject.getByText(BusinessFormationDefaults.previousButtonText));
       await waitFor(() => {
         expect(subject.queryByTestId("business-section")).toBeInTheDocument();
       });
     });
 
-    it("navigates back to contact tab from the payment tab", async () => {
+    it("routes to profile page when edit business name button is clicked", async () => {
       renderWithData({});
-      await submitBusinessTab();
-      await submitContactsTab();
-      fireEvent.click(subject.getByText(BusinessFormationDefaults.previousButtonText));
+      fireEvent.click(subject.getByTestId("edit-business-name"));
+      expect(mockPush).toHaveBeenCalledWith("/profile");
+    });
 
-      await waitFor(() => {
-        expect(subject.queryByTestId("contacts-section")).toBeInTheDocument();
+    it("routes to profile page when edit legal structure button is clicked", async () => {
+      renderWithData({});
+      fireEvent.click(subject.getByTestId("edit-legal-structure"));
+      expect(mockPush).toHaveBeenCalledWith("/profile");
+    });
+
+    describe("navigates from the review page", () => {
+      it("displays the first tab when the edit button in the main business section is clicked", async () => {
+        renderWithData({});
+        await submitBusinessTab();
+        await submitContactsTab();
+        fireEvent.click(subject.getByTestId("edit-business-name-section"));
```

Comment:
if possible (seems it is in this case) I'd prefer select by text, using the configured text (here, `BusinessFormationDefaults.editButtonText`)

Reply:
applies to several tests here, won't comment on all of them

Reply:
There are 7 tests that I wrote that use "fireEvent.click(subject.getByTestId....)". They all share `BusinessFormationDefaults.editButtonText` as the content. I could use subject.getAllByText, however, I'll need to state the index within the array. I don't think this is a good idea.

Reply:
I'll look into grabbing the input by the aria label

Reply:
Ohh I see your point. Didn't realize there were multiple elements with the same text. In that case, I wouldn't bother changing it. Sticking with testid seems perfectly reasonable